src/harness/testcases/WINNF_FT_S_FDB_testcase.py

```diff
@@ -51,993 +34,148 @@ def tearDown(self):
-                      expected_response_codes)
-      if response['response']['responseCode'] == 0:  # SUCCESS
-        self.assertTrue('cbsdId' in response)
-        cbsd_ids.append(response['cbsdId'])
-      else:
-        self.assertFalse('cbsdId' in response)
-
-    del request, responses
-    config['grantRequests'][0]['cbsdId'] = cbsd_ids[0]
-    request = {'grantRequest': config['grantRequests']}
-    responses = self._sas.Grant(request)['grantResponse']
-    # Check Grant responses.
-    for i in range(len(responses)):
-      response = responses[i]
-      expected_response_codes = config['expectedGrantResponseCodes']
-      logging.debug('Expecting to see response code in set %s in response: %s',
-                      expected_response_codes, response)
-      self.assertIn(response['response']['responseCode'],
-                      expected_response_codes)
-      if response['response']['responseCode'] == 0:  # SUCCESS
-          self.assertTrue('cbsdId' in response)
-          self.assertTrue('grantId' in response)
-          grant_expire_time = datetime.strptime(response['grantExpireTime'],
-                                                '%Y-%m-%dT%H:%M:%SZ')
-      else:
-          self.assertFalse('cbsdId' in response)
-          self.assertFalse('grantId' in response)
-
-    # Check grant response
-    self.assertEqual(responses[0]['cbsdId'], cbsd_ids[0])
-    grant_id = responses[0]['grantId']
-    grant_expire_time = datetime.strptime(responses[0]['grantExpireTime'],
-                                          '%Y-%m-%dT%H:%M:%SZ')
-    del request, responses
-    # Load exclusion zone database which includes a zone that contains the CBSD location or is within 50 meters of the CBSD location
-    self._sas_admin.InjectDatabase_url(config['dpa_movelist_db'])
+    #Create fake database server
+    self._fake_database_server = FakeDatabaseTestHarness( \
+      config['fake_fcc_db']['url'], config['fake_fcc_db']['databaseFile'] )
+    #Start fake database server
+    self._fake_database_server.start()
+
+    #Inject FCC ID and User Id of the device into UUT.    
+    fcc_id = (config['registrationRequest'][0]['fccId'], 47) 
+    user_id = config['registrationRequest'][0]['userId']
+    self._sas_admin.InjectFccId({
+      'fccId': fcc_id[0],
+      'fccMaxEirp': fcc_id[1]
+    })
+    self._sas_admin.InjectUserId({'userId': user_id})
+
+    cbsd_ids, grant_ids = self.assertRegisteredAndGranted(config['registrationRequest'], config['grantRequest'])
+   
+    #Loading the FCC database URL into the UUT, the database has a zone containing the CBSD location or is within 50 meters of the CBSD location
+    self._sas_admin.InjectDatabaseUrl(config['fake_fcc_db']['url'])
 
     # Trigger daily activities
     self.TriggerDailyActivitiesImmediatelyAndWaitUntilComplete()
 
-    # Response code 500(TERMINATED_GRANT)or 501 (SUSPENDED_GRANT) should be sent for Heartbeat Request
+    #Construct heartbeat message
     request = {
```

Comment:
You need to handle multiple CBSDs+Grants (here and below).

Reply:
Added CatB device   and sending multiple registration and grants